src/basic/MAT_FUSED_MUL_ADD.hpp

```diff
@@ -0,0 +1,91 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Compute D = A x B + C, where
+// Inputs:
+// A: N/(Ne*Ne) Ne x Ne matrices
+// B: N/(Ne*Ne) Ne x Ne matrices
+// Ouput:
+// D: N/(Ne*Ne) Ne x Ne matrices
+// All square row-major matrices, C is ignored.
+//for(Index_type ii = 0; ii != (N/(Ne*Ne)); ++ii){
+//      for(Index_type row = 0; row != Ne; ++row){
+//        for(Index_type col = 0; col != Ne; ++col){
+//            MAT_FUSED_MUL_ADD_BODY;
+//        }   
+//    }   
+//} 
+#ifndef RAJAPerf_Basic_MAT_FUSED_MUL_ADD_HPP
+#define RAJAPerf_Basic_MAT_FUSED_MUL_ADD_HPP
+
+#include "RAJA/RAJA.hpp"
+#include "common/KernelBase.hpp"
+
+#define MAT_FUSED_MUL_ADD_DATA_INIT                                                  \
+for(Index_type ii = 0; ii != (N/(Ne*Ne)); ++ii){                                     \
+	for(Index_type i = 0; i != NeNe; ++i){ m_A[i+(ii*NeNe)] = i; }               \
+	for(Index_type i = 0; i != NeNe; ++i){ m_B[i+(ii*NeNe)] = NeNe - 1 - i; }    \
+} 
+
+#define MAT_FUSED_MUL_ADD_DATA_SETUP                                                 \
+  Real_ptr A = m_A; 						                     \
+  Real_ptr B = m_B; 								     \
+  Real_ptr D = m_D; 						
+
+#define MAT_FUSED_MUL_ADD_BODY                                               	     \
+  Real_type dot = 0;                                                         	     \
+  for (Index_type k = 0; k < Ne; ++k) {                                      	     \
+    dot += A[row*Ne + k + ii*(Ne*Ne)] * B[k*Ne + col + ii*(Ne*Ne)];          	     \
+  }                                                                          	     \
+  D[row*Ne + col + ii*(Ne*Ne)] = dot;                                        	     \
```

Comment:
Is this supposed to be doing `D = A*B` or `d = A*B+C`?

Reply:
The mfma instructions are for computing D = A x B + C but we're assuming for this case C is zeros ignored. It's defined this way so we can expand future cases with a non-trivial C matrix.

Reply:
The name of the kernel is confusing then if we're not actually doing the ADD part.